activerecord/test/cases/relation/with_test.rb

```diff
@@ -57,9 +58,28 @@ def test_with_when_called_from_active_record_scope
       end
 
       def test_with_when_invalid_params_are_passed
-        assert_raise(ArgumentError) { Post.with }
         assert_raise(ArgumentError) { Post.with(posts_with_tags: nil).load }
-        assert_raise(ArgumentError) { Post.with(posts_with_tags: [Post.where("tags_count > 0")]).load }
+        assert_raise(ArgumentError) { Post.with(posts_with_tags: [Post.where("tags_count > 0"), 5]).load }
+      end
+
+      def test_with_when_passing_arrays
+        relation = Post
+          .with(posts_with_tags_or_comments: [
+            Post.where("tags_count > 0"),
+            Post.where("legacy_comments_count > 0")
+          ])
+          .from("posts_with_tags_or_comments AS posts")
+
+        assert_equal POSTS_WITH_TAGS_OR_COMMENTS, relation.order(:id).pluck(:id)
```

Comment:
This actually fails on sqlite3 with `ActiveRecord::StatementInvalid: SQLite3::SQLException: near "(": syntax error`.

I believe this is because of <https://github.com/rails/rails/pull/51549>, which wraps the `SELECT` in parentheses, which appears to be legal with PostgreSQL but not SQLite3.

Reply:
Addressed indeed!